eventsourcing/src/test/scala/com/evolutiongaming/akkaeffect/eventsourcing/EngineTestCases.scala

```diff
@@ -405,6 +409,49 @@ abstract class EngineTestCases extends AsyncFunSuite with Matchers {
     result.use { _.pure[F] }
   }
 
+  def `effective state should not change after persistence failed`[F[_]: Async: ToFuture: FromFuture]
+    : F[Unit] = {
+    val error = new RuntimeException()
```

Comment:
`RuntimeException with NoStackTrace` please

Reply:
its just a test, does it matter?

Reply:
why do I need `NoStackTrace` in tests?

Reply:
because collecting stack trace is a heavy operation, and you don't care about stacktrace in this test

Reply:
I don't care about heavy operations in tests as well. Especially if the exception not thrown

Reply:
stacktrace collected when you call `new` and not when `throw`

Reply:
just check what `NoStackTrace` does and usage of `fillInStackTrace` :)